Server/src/main/java/org/gluu/oxauth/session/ws/rs/EndSessionRestWebServiceImpl.java

```diff
@@ -146,15 +146,8 @@ public Response requestEndSession(String idTokenHint, String postLogoutRedirectU
                     continue;
                 }
 
-                for (String logoutUri : client.getFrontChannelLogoutUri()) {
-                    if (Util.isNullOrEmpty(logoutUri)) {
-                        continue; // skip if logout_uri is blank
-                    }
-
-                    if (client.getFrontChannelLogoutSessionRequired()) {
-                        logoutUri = EndSessionUtils.appendSid(logoutUri, pair.getFirst().getOutsideSid(), appConfiguration.getIssuer());
-                    }
-                    frontchannelUris.add(logoutUri);
+                if(StringUtils.isNotBlank(client.getFrontChannelLogoutUri())) {
+                    frontchannelUris.add(EndSessionUtils.appendSid(client.getFrontChannelLogoutUri(), pair.getFirst().getOutsideSid(), appConfiguration.getIssuer()));
```

Comment:
sid should appended only if `frontChannelLogoutSessionRequired=true`.
```
                    logoutUri = client.getFrontChannelLogoutUri();
                    if (client.getFrontChannelLogoutSessionRequired()) {	
                        logoutUri = EndSessionUtils.appendSid(logoutUri, pair.getFirst().getOutsideSid(), appConfiguration.getIssuer());	
                    }	
                    frontchannelUris.add(logoutUri);
```

Reply:
Done